### PR TITLE
Feature/implementing spring aop

### PR DESCRIPTION
**Pull request by yavlash-liza**
Implementing Spring AOP

### CODE REVIEW

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.context.annotation.Configuration;
+
+@Slf4j
+@Aspect
+@Configuration
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:11]: what are you going to configure in this class????

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class LoggingAfterSuccessFindAndGetRepositoryAspect {
+
+    @AfterReturning(value = "execution(* by.library.yavlash.repository..*find*(..))"
+            + "|| execution(* by.library.yavlash.repository..*get*(..))")
+    public void logStatementAfter(JoinPoint joinPoint) {
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:16]: it's not easy to understand what this method does. try to rename it

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        log.info("Target method successfully completed: {}", joinPoint.getStaticPart().getSignature());
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:17]: you can add some information about method, arguments, class name. Please add variables when you remake this aspect.  You should take the main message of text out of the method into static final variable

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.context.annotation.Configuration;
+
+@Slf4j
+@Aspect
+@Configuration
+public class LoggingAfterSuccessFindAndGetRepositoryAspect {
+
+    @AfterReturning(value = "execution(* by.library.yavlash.repository..*find*(..))"
+            + "|| execution(* by.library.yavlash.repository..*get*(..))")
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:15]: the script is written incorrectly in "* by.library.yavlash.repository..*find*(..))" and "* by.library.yavlash.repository..*get*(..))"

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.context.annotation.Configuration;
+
+@Slf4j
+@Aspect
+@Configuration
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java:11]: the same question as above

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+public class LoggingExceptionInRepositoryAspect {
+
+    @Around(value = "execution(* by.library.yavlash.repository.*.*(..))")
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java:14]: it's incorrect choice of aspect strategy. But script is written right

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    public Object logStatementAfter(ProceedingJoinPoint joinPoint) throws Throwable {
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java:15]: there is incorrect parameters of method

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        try {
+            return joinPoint.proceed();
+        } catch (Exception exception) {
+            log.error("Exception {} ", exception.getMessage());
+            throw new Exception(exception.getMessage());
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java:20]: when you remake the comment above. you will not use try catch statement. You can log method name, class name, arguments, exception message when you create your custom log message

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java (hunk 1 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:11]: without this annotation code didn't worked

Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:11]: do you think I added it without checking?

Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:11]: done

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java (hunk 2 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:16]: done

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java (hunk 5 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java:11]: done

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java (hunk 6 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java:14]: done

File: src/main/java/by/library/yavlash/aspect/AppConfig.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package by.library.yavlash.aspect;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+
+@Configuration
+@EnableAspectJAutoProxy
+public class AppConfig {
+}
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aspect/AppConfig.java:9]: you shouldn't use this class at all. remove it

File: src/main/java/by/library/yavlash/aspect/AfterSuccessFindAndGetRepositoryAspect.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+
+@Slf4j
+@Aspect
+public class AfterSuccessFindAndGetRepositoryAspect {
+    private static final String message = "Target method successfully completed: {}";
+
+    @AfterReturning(value = "execution(* by.library.yavlash.repository.*find*(..))"
+            + "|| execution(* by.library.yavlash.repository.*get*(..))")
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aspect/AfterSuccessFindAndGetRepositoryAspect.java:14]: this script is incorrect. you should write an aspect which will log methods starting with find or get

File: src/main/java/by/library/yavlash/aspect/ExceptionInRepositoryAspect.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import org.aspectj.lang.annotation.AfterThrowing;
+import org.aspectj.lang.annotation.Aspect;
+
+@Slf4j
+@Aspect
+public class ExceptionInRepositoryAspect {
+    private static final String message = "Exception {} ";
+
+    @AfterThrowing(value = "execution(* by.library.yavlash.repository.*.*(..))")
+    public Object logStatementAfterExceptionInRepository(ProceedingJoinPoint joinPoint) throws Throwable {
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aspect/ExceptionInRepositoryAspect.java:14]: you need to do research more to find other examples of realization method which handle throwing exceptions

File: src/main/java/by/library/yavlash/aspect/ExceptionInRepositoryAspect.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+        try {
+            return joinPoint.proceed();
+        } catch (Exception exception) {
+            log.error(message, exception.getMessage());
+            throw new Exception(exception.getMessage());
+        }
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aspect/ExceptionInRepositoryAspect.java:20]: you should use a try catch statement implementing this method

File: src/main/java/by/library/yavlash/aspect/AfterSuccessFindAndGetRepositoryAspect.java [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    public void logStatementAfterSuccessRepositoryFindOrGetMethod(JoinPoint joinPoint) {
```
Comment (kolyago-vladislav) [inline src/main/java/by/library/yavlash/aspect/AfterSuccessFindAndGetRepositoryAspect.java:15]: you need to do research more to find other examples of realization the method like this

File: src/main/java/by/library/yavlash/aspect/AppConfig.java (hunk 9 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aspect/AppConfig.java:9]: done

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java (hunk 4 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java:15]: done

File: src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java (hunk 3 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingAfterSuccessFindAndGetRepositoryAspect.java]: done

File: src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java (hunk 7 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aop/LoggingExceptionInRepositoryAspect.java]: done

File: src/main/java/by/library/yavlash/aspect/ExceptionInRepositoryAspect.java (hunk 12 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aspect/ExceptionInRepositoryAspect.java]: done

File: src/main/java/by/library/yavlash/aspect/AfterSuccessFindAndGetRepositoryAspect.java (hunk 13 above)
Comment (yavlash-liza) [reply src/main/java/by/library/yavlash/aspect/AfterSuccessFindAndGetRepositoryAspect.java]: done

**Review by kolyago-vladislav — APPROVED** [review]

**Pull request merged**